packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -216,6 +216,7 @@ class HotRunner extends ResidentRunner {
       // Measure time to perform a hot restart.
       printStatus('Benchmarking hot restart');
       await restart(fullRestart: true);
+      await Future<void>.delayed(const Duration(seconds: 2));
```

Comment:
So same question here - can we wait for something to get printed to stdout confirming that restart has finished instead of a delay?  I'm worried that this will become just as flaky as our current setup, and then we bump it to 4 or 10 or 20 seconds

Reply:
The issue is that the restart has finished as far as we know, and there is no race condition in hot restart. The set of stream extensions are extensible and can come in any order. Another issue might be a potential GC, so regardless we should be yielding in some way.

I understand the concerns here, the benchmarks should be as deterministic as possible. But in the context of two machines talking over rpc in a garbage collected language, I'm not sure that is possible in a reasonable way